integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r3/TaskFhirResourceProviderIntegrationTest.java

```diff
@@ -0,0 +1,423 @@
+  assertThat(response.getContentAsString(), notNullValue());
+  
+  OperationOutcome operationOutcome = readOperationOutcome(response);
+  
+  assertThat(operationOutcome, notNullValue());
+  assertThat(operationOutcome.hasIssue(), is(true));
+ }
+ 
+  
+	@Test
+	public void shouldReturnNotFoundWhenUpdatingNonExistentTaskAsXML() throws Exception {
+		// get the existing record
+		MockHttpServletResponse response = get("/Task/" + TASK_UUID).accept(FhirMediaTypes.XML).go();
+		Task task = readResponse(response);
+		
+		// update the existing record
+		task.setId(WRONG_TASK_UUID);
+		
+		// send the update to the server
+		response = put("/Task/" + TASK_UUID).xmlContext(toXML(task)).go();
+		
+		assertThat(response, isNotFound());
+		assertThat(response.getContentType(), is(FhirMediaTypes.XML.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		OperationOutcome operationOutcome = readOperationOutcome(response);
+		
+		assertThat(operationOutcome, notNullValue());
+		assertThat(operationOutcome.hasIssue(), is(true));
+		
+	}
+ @Test
+ public void shouldReturnNotFoundWhenUpdatingNonExistentTaskAsJSON() throws Exception {
+  // get the existing record
+    MockHttpServletResponse response = get("/Task/" + TASK_UUID).accept(FhirMediaTypes.JSON).go();
+    Task task = readResponse(response);
+    
+    // update the existing record
+    task.setId(WRONG_TASK_UUID);
+    
+    // send the update to the server
+    response = put("/Task/" + WRONG_TASK_UUID).jsonContent(toJson(task)).jsonContent(toJson(task)).go();
+    
+    assertThat(response, isNotFound());
+    assertThat(response.getContentType(), is(FhirMediaTypes.JSON.toString()));
+    assertThat(response.getContentAsString(), notNullValue());
+    
+    OperationOutcome operationOutcome = readOperationOutcome(response);
+    
+    assertThat(operationOutcome, notNullValue());
+    assertThat(operationOutcome.hasIssue(), is(true));
+ }
+	
+	@Test
+	public void shouldDeleteExistingTask() throws Exception {
+		MockHttpServletResponse response = delete("/Task/" + TASK_UUID).accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, isOk());
+		
+		response = get("/Task/" + TASK_UUID).accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, statusEquals(HttpStatus.GONE));
+	}
+	
+	@Test
+	public void shouldReturnNotFoundWhenDeletingNonExistentTask() throws Exception {
+		MockHttpServletResponse response = delete("/Patient/" + WRONG_TASK_UUID).accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, isNotFound());
+		assertThat(response.getContentType(), is(FhirMediaTypes.JSON.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		OperationOutcome operationOutcome = readOperationOutcome(response);
+		
+		assertThat(operationOutcome, notNullValue());
+		assertThat(operationOutcome.hasIssue(), is(true));
+	}
+	
+	@Test
+	public void shouldSearchForExistingTasksAsJson() throws Exception {
```

Comment:
This is actually a search for all tasks. We need one additional method to search for Tasks using filtering criteria, just to ensure that part works as well. Something like a GET request to the URI `/Task/?status=accepted&_sort=_lastUpdated`.

Reply:
ok let me do this now thanks

Reply:
Resolved

Reply:
@sherrif10 I don't see a relevant code change here? Can you ensure you've pushed your latest changes please?